netman/adapters/switches/remote.py

```diff
@@ -38,6 +38,8 @@ def __init__(self, switch_descriptor):
         super(RemoteSwitch, self).__init__(switch_descriptor)
         self.requests = requests
         self.session_id = None
+        self.request_url_key = 'url'
```

Comment:
I think that if the flask test client does not comply 100% to requests we should'nt have injected cross dependent stuff like that. 

May I propose a sort of HttpProvider interface with a default implementation using requests and the test will use another implementation for the test.

Another suggestion is to wrap the flask test client in a wrapper that complies to requests

I think the 2nd is better